### PR TITLE
Add support for the 2022 entity model in HA

### PR DESCRIPTION
**Pull request by tichris0**
The new HA entity model has changed the way unique IDs work.  Before we create the objects, we need the serial number to bake that as part of the unique ID.  This change adds an async function to know when the required information is available prior to entity creation.

Overall, this enables the AlarmDecoder component to support the new modern infrastructure which creates a nicer hierarchical model that the user can now browse, operate on and etc.

This change requires a bump in the python version to 3.12 (which is now required by HA).  That's a bit tricky because Ubuntu doesn't yet seem to have a 3.12 apt-get package readily available, although it's quite easy to install

"sudo add-apt-repository ppa:deadsnakes/ppa && sudo apt update && sudo apt install python3.12"

### CODE REVIEW

File: .vs/slnx.sqlite
Comment (ajschmidt8) [inline .vs/slnx.sqlite:1]: can you remove this file from the PR?

File: .vs/ProjectSettings.json
Comment (ajschmidt8) [inline .vs/ProjectSettings.json:1]: can you remove this file from the PR?

**Review by ajschmidt8 — COMMENTED** [review]
is there any Home Assistant documentation related to these changes that you can link?

File: .vs/slnx.sqlite
Comment (tichris0) [reply .vs/slnx.sqlite:1]: Yes of course. done!

File: .vs/ProjectSettings.json
Comment (tichris0) [reply .vs/ProjectSettings.json:1]: Yes of course. done!

**Comment by tichris0** [comment]
> is there any Home Assistant documentation related to these changes that you can link?

Here's the HA documentation on the _has_entity_name_ which is mandatory for new integrations #https://developers.home-assistant.io/docs/core/entity/#entity-naming.  Currently the alarmdecoder component is on life-support and doesn't play well with the entity model that was introduced a couple years back.  It pairs with unique_id which is where this change comes in.  I need the device's serial number to generate that unique_id for the domain.

Here's the bug that I have a fix for: #https://github.com/home-assistant/core/issues/79599.  It best illustrates what the issue is when a component is stuck using the deprecated entity model.

**Comment by tichris0** [comment]
Hi @ajschmidt8 , any additional thoughts on the latest PR?  I'd like to move forward with the HA change as well assuming this gets approval to move forward.

**Review by ajschmidt8 — APPROVED** [review]

**Comment by ajschmidt8** [comment]
released here: https://pypi.org/project/adext/0.4.3/

**Pull request merged**